ui/nuxeo-document-picker/nuxeo-document-picker.js

```diff
@@ -0,0 +1,219 @@
+            max-height: 84%;
+            max-width: 768px;
+            padding: 8px;
+            overflow: auto;
+          }
+          #topContainer {
+            display: flex;
+            justify-content: flex-end;
+          }
+          iron-icon {
+            opacity: 0.6;
+          }
+          iron-icon:hover {
+            cursor: pointer;
+            opacity: 1;
+          }
+          #mainContainer,
+          #topContainer,
+          nuxeo-results-view {
+            margin: 0;
+            padding: 0;
+          }
+          nuxeo-results-view {
+            --nuxeo-results-view-height: calc(100vh - 520px - var(--nuxeo-app-top));
+          }
+        </style>
+        <nuxeo-dialog id="dialog" modal>
+          <div id="mainContainer">
+            <div id="topContainer">
+              <iron-icon icon="nuxeo:cross" on-tap="close"></iron-icon>
+            </div>
+            <nuxeo-results-view
+              id="resultsView"
+              provider="[[provider]]"
+              page-size="[[pageSize]]"
+              params="[[_params]]"
+              quick-filters="{{_quickFilters}}"
+              schemas="[[schemas]]"
+              enrichers="[[enrichers]]"
+              search-name="[[searchName]]"
+              aggregations="{{_aggregations}}"
+              href-base="[[hrefBase]]"
+              visible
+              show-filters
+              opened
+              on-navigate="_onNavigate"
+              on-results-changed="_onResultsChanged"
+            ></nuxeo-results-view>
+          </div>
+          <div class="buttons">
+            <!-- TODO check if we can have these stuck to the bottom (absolute positioning? check creation wizard) -->
+            <paper-button noink class="secondary" dialog-dismiss>[[i18n('command.cancel')]]</paper-button>
+            <paper-button noink class="primary" on-tap="_onSelect" id="selectButton">
+              [[i18n('command.select')]]
+            </paper-button>
+          </div>
+        </nuxeo-dialog>
+      `;
+    }
+
+    static get is() {
+      return 'nuxeo-document-picker';
+    }
+
+    static get properties() {
+      return {
+        /**
+         * List of content enrichers passed on to `provider`.
+         */
+        enrichers: String,
+        /**
+         * The base url for the hrefs of the search layouts to be loaded.
+         */
+        hrefBase: String,
+        /**
+         * The number of results per page.
+         */
+        pageSize: Number,
+        /**
+         * The id of the `nuxeo-page-provider` instance used to perform the search.
+         */
+        provider: String,
+        /**
+         * List of comma separated values of the document schemas to be returned.
+         * All document schemas are returned by default.
+         */
+        schemas: String,
+        /**
+         * The name of the search layout.
+         */
+        searchName: String,
```

Comment:
I don't know, even though the default use case uses the same value for `searchName` and `provider`, I chose to keep both to maintain the same API as `<nuxeo-search-form-layout>`, `<nuxeo-search-results-layout>` and `<nuxeo-results-view>`.
I wouldn't be against having only `provider` here though, since we could easily add `searchName` in the future if we wanted to.